hw/ip/otp_ctrl/doc/_index.md

```diff
@@ -8,14 +8,285 @@ title: "OTP Controller Technical Specification"
+HW_CFG         | Hardware configuration     | no     | yes      | yes (digest) |      no      | Hardware configuration bits used to hardwire specific hardware functionality. <br> E.g., raw entropy accessibility or flash scrambling bypass range.
+LIFE_CYCLE     | Life-cycle state, counters | no     | yes      |     no       |      no      | Life-cycle related bits (ADD link). **Note**, this partition cannot be locked as the life-cycle state needs to be able to advance to RMA in the field.
+SECRET         | Tokens and Keys            | yes    | yes      | yes (digest) | yes (digest) | Device-unique secrets.
+
+Generally speaking, the production life cycle of a device is split into 5 stages "Manufacturing" -> "Calibration and Testing" -> "Provisioning" -> "Mission" -> "RMA".
+OTP values are usually programmed during "Calibration and Testing", "Provisioning" and "RMA" stages, as explained below.
+A detailed listing of all the collateral and the corresponding memory map can be found in the Programmer's Guide Section further below.
+
+### Calibration and Test
+
+During this stage, the device is tested for functionality and calibrated to ensure uniformity.
+The calibration can focus on a number of things, but usually is centered around adjusting clock, voltage and timing sources to remove process variation.
+These calibration values are programmed into the CREATOR_SW_CFG partition, as they are non-secret values meant to be read out by software and programmed into respective peripherals.
+
+Early on during this stage, the various secret unlock tokens are also programmed into the secret partition and harvested by the silicon creator.
+
+### Provisioning
+
+During this stage, the device is provisioned with the final firmware and a "unique" seed or identity.
+The secret partition is populated with root secrets and keys that are critical to establishing the device identity.
+
+As part of injecting the final firmware, the SKU-specific hardware and software configurations are also programmed.
+
+### Life Cycle Partition
+
+The life cycle partition is active throughout all stages and hence it is the **ONLY** partition that cannot be locked.
+After the device finishes provisioning and goes into production, it must retain the ability to transition back to RMA in case of unexpected failures.
+
+In order to support this transition, the life-cycle state and counters must always be update-able (**TODO: add link to life-cycle docs for more info**).
+
+## Locking a Partition
+
+Write access to a partition can be permanently locked when software determines it will no longer make any updates to that partition.
+To lock, a truncated integrity constant (32bit) is calculated and programmed alongside the other data of that partition.
+
+Once the "integrity digest" is non-zero, no further updates are allowed.
+If the partition is secret, software is in addition no longer able to read the contents (see secret partiton description further below).
```

Comment:
maybe add a soft link for "secret partition description" . `partiton` -> `partition`

Reply:
good idea, will do